pkg/puller/watcher.go

```diff
@@ -0,0 +1,126 @@
+package puller
+
+import (
+	"encoding/json"
+	"github.com/fsnotify/fsnotify"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+)
+
+var p *Puller
+
+type Puller struct {
+	modelDir string
+	onConfigChange func(EventWrapper)
+	fileExtension string
+}
+
+type ModelDefinition struct {
+	// TODO: This needs to be defined by the ConfigMap PR
+	StorageUri string  `json:"storageUri"`
+	Framework string  `json:"framework"`
+	Memory string  `json:"memory"`
+}
+
+func init() {
+	p = New()
+}
+
+func New() *Puller {
+	p := new(Puller)
+	// TODO: This should probably be overridable
+	p.fileExtension = ".json"
+	return p
+}
+
+type State string
+
+const (
+	// State Related
+	ShouldLoad State = "Load"
+	ShouldUnload State = "Unload"
+
+ 	writeOrCreateMask = fsnotify.Write | fsnotify.Create
```

Comment:
What does the writeOrCreateMask state mean? Why is it different from the ShouldLoad state?

Reply:
nvm, I saw how it is used later  👍